projects/coolstore-kjar-s2i/Jenkinsfile

```diff
@@ -106,9 +114,90 @@ node {
 }
 
 def processStageResult() {
-//    step([$class: 'Mailer', notifyEveryUnstableBuild: true, recipients: "${EMAIL}", sendToIndividuals: false])
 
-    if (currentBuild.result != null && currentBuild.result != "SUCCESS") {
+    if (currentBuild.result != null && currentBuild.result == "FAILURE") {
         sh "exit 1"
     }
 }
+
+def login() {
+    sh """
+       set +x
+       oc login --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --token=\$(cat /var/run/secrets/kubernetes.io/serviceaccount/token) https://kubernetes.default.svc.cluster.local >/dev/null 2>&1 || echo 'OpenShift login failed'
+       """
+}
+
+def parseVersion(String sourceFileName, String destinationFileName) {
+    sh """
+    set +x
+   full_version=\$(cat $sourceFileName)
+   parsed_version=\${full_version:33}
+   mkdir -p \$(dirname ${destinationFileName})
+   echo \$parsed_version > ${destinationFileName}
+    """
+    
+    return readFile("${destinationFileName}").trim()
+}
+
+def parseImageVersion(String sourceFileName, String destinationFileName, String rulesVersion, String buildVersion) {
```

Comment:
I battled with this method for 6hrs and it went through a bunch of refactoring. we can remove that parameter